### PR TITLE
Feature/468 operator naming

### PR DESCRIPTION
**Pull request by todderasesareddot**

### CODE REVIEW

File: Python/kraken/core/configs/config.py [hunk 1]
```diff
@@ -326,6 +326,8 @@ def initNameTemplate(self):
                     "Layer": ["container", "sep", "name"],
                     "ComponentGroup": ["name", "sep", "location", "sep", "type"],
                     "default": ["component", "sep", "location", "sep", "name", "sep", "type"],
+                    "KLOperator": ["component", "sep", "location", "sep", "name", "sep", "solverSource", "sep", "solverName", "sep", "type"],
```
Comment (EricTRocks) [inline Python/kraken/core/configs/config.py]: This is a bit risky as it means that other studios who don't want this naming on their ops will have the op names changed. Naming should stay the same in the core config, but be able to be changed in studio specific configs. It's OK in this change I believe but need to watch out for this stuff in the future.

File: Python/kraken/core/objects/operators/operator.py [hunk 2]
```diff
@@ -97,6 +97,18 @@ def getBuildName(self):
...
+                if self.isTypeOf("KLOperator"):
+                    builtName += self.solverTypeName
+                else:
+                    builtName += self.canvasPresetPath.rpartition('.')[-1]
+
+            elif token is 'solverSource':
+                if self.isTypeOf("KLOperator"):
+                    builtName += self.extension
+                else:
+                    builtName += self.canvasPresetPath.rpartition('.')[0].replace('.', '_')
```
Comment (EricTRocks) [inline Python/kraken/core/objects/operators/operator.py]: Is this safe to put in underscores? What if the naming convention at a studio doesn't use underscores?

**Comment by todderasesareddot** [comment]
Hmmm..  I guess we can just strip out the periods?

**Comment by EricTRocks** [comment]
Yeah I think that would be the best solution.

File: Python/kraken/core/objects/operators/operator.py (hunk 2 above)
Comment (todderasesareddot) [inline Python/kraken/core/objects/operators/operator.py]: OK, just strip out non alphanumeric?

**Comment by EricTRocks** [comment]
Stripping out all non-alpha-numeric is good.

File: Python/kraken/core/objects/operators/operator.py (hunk 2 above)
Comment (todderasesareddot) [inline Python/kraken/core/objects/operators/operator.py]: Also, thought about if pipeline requires all lowercase or something.  Is there a way to force lowercase for all tokens?

**Comment by EricTRocks** [comment]
Looks good!

**Comment by todderasesareddot** [comment]
I updated it to remove all non-alphanumeric, but you may have committed before I did!

**Comment by EricTRocks** [comment]
Oops. Open another pull request. :)

**Pull request merged**